### PR TITLE
Fixed #34140 -- Reformatted code blocks in docs with blacken-docs.

### PR DESCRIPTION
**Pull request by felixxm**
- [x] Added `black` rule to `make.bat`
- [x] Added pre-commit hook
- [x] Update contributing guides

### CODE REVIEW

**Comment by smithdc1** [comment]
Will the `make.bat` file be updated as well?

Let me know if you need help with that?

**Comment by felixxm** [comment]
> Will the `make.bat` file be updated as well?
> 
> Let me know if you need help with that?

Yes, it's on my TO-DO list, and yes, I need help :ambulance:  :wink:

File: docs/Makefile [hunk 1]
```diff
@@ -49,6 +49,7 @@ help:
 	@echo "  linkcheck  to check all external links for integrity"
 	@echo "  doctest    to run all doctests embedded in the documentation (if enabled)"
 	@echo "  spelling   to check for typos in documentation"
+	@echo "  blacken    to apply the black formatting to code blocks in documentation"
```
Comment (smithdc1) [inline docs/Makefile:52]:
```suggestion
	@echo "  black    to apply the black formatting to code blocks in documentation"
```

**Comment by smithdc1** [comment]
This is what I've come up with so far. I'm assuming the error code doesn't matter (`mkdir` will set `%ERRORLEVEL%` if the folder already exists).

```diff
diff --git a/docs/make.bat b/docs/make.bat
index 4743692ca3..1de9089b35 100644
--- a/docs/make.bat
+++ b/docs/make.bat
@@ -35,6 +35,7 @@ if "%1" == "help" (
        echo.  linkcheck  to check all external links for integrity
        echo.  doctest    to run all doctests embedded in the documentation if enabled
        echo.  spelling   to check for typos in documentation
+       echo.  black      to apply the black formatting to code blocks in documentation 
        goto end
 )

@@ -196,4 +197,11 @@ spelling/output.txt.
        goto end
 )

+if "%1" == "black" (
+       mkdir %BUILDDIR%
+       for /f "usebackq tokens=*" %%i in (`dir *.txt /s /b ^| findstr /v /c:"_build" /c:"_theme"`) do (
+               blacken-docs --rst-literal-block %%i
+       )
+       echo "Code blocks reformatted"
+)
+
```

Running blacken-docs it creates one diff to this PR:

<details>

```diff 

diff --git a/docs/ref/contrib/postgres/search.txt b/docs/ref/contrib/postgres/search.txt
index 699f81bd11..4f481e56a2 100644
--- a/docs/ref/contrib/postgres/search.txt
+++ b/docs/ref/contrib/postgres/search.txt
@@ -343,9 +343,7 @@ Usage example:
     >>> Author.objects.create(name="Katy Stevens")
     >>> Author.objects.create(name="Stephen Keats")
     >>> test = "Katie Stephens"
-    >>> Author.objects.annotate(
-    ...     similarity=TrigramSimilarity("name", test),
-    ... ).filter(
+    >>> Author.objects.annotate(similarity=TrigramSimilarity("name", test),).filter(
     ...     similarity__gt=0.3
     ... ).order_by("-similarity")
     [<Author: Katy Stevens>, <Author: Stephen Keats>]
@@ -366,9 +364,7 @@ Usage example:
     >>> Author.objects.create(name="Katy Stevens")
     >>> Author.objects.create(name="Stephen Keats")
     >>> test = "Kat"
-    >>> Author.objects.annotate(
-    ...     similarity=TrigramWordSimilarity(test, "name"),
-    ... ).filter(
+    >>> Author.objects.annotate(similarity=TrigramWordSimilarity(test, "name"),).filter(
     ...     similarity__gt=0.3
     ... ).order_by("-similarity")
     [<Author: Katy Stevens>]
@@ -401,9 +397,7 @@ Usage example:
     >>> Author.objects.create(name="Katy Stevens")
     >>> Author.objects.create(name="Stephen Keats")
     >>> test = "Katie Stephens"
-    >>> Author.objects.annotate(
-    ...     distance=TrigramDistance("name", test),
-    ... ).filter(
+    >>> Author.objects.annotate(distance=TrigramDistance("name", test),).filter(
     ...     distance__lte=0.7
     ... ).order_by("distance")
     [<Author: Katy Stevens>, <Author: Stephen Keats>]
@@ -424,9 +418,7 @@ Usage example:
     >>> Author.objects.create(name="Katy Stevens")
     >>> Author.objects.create(name="Stephen Keats")
     >>> test = "Kat"
-    >>> Author.objects.annotate(
-    ...     distance=TrigramWordDistance(test, "name"),
-    ... ).filter(
+    >>> Author.objects.annotate(distance=TrigramWordDistance(test, "name"),).filter(
     ...     distance__lte=0.7
     ... ).order_by("distance")
     [<Author: Katy Stevens>]

```

</details>

**Comment by felixxm** [comment]
> This is what I've come up with so far. I'm assuming the error code doesn't matter (`mkdir` will set `%ERRORLEVEL%` if the folder already exists).
> 
> ```diff
> diff --git a/docs/make.bat b/docs/make.bat
> index 4743692ca3..1de9089b35 100644
> --- a/docs/make.bat
> +++ b/docs/make.bat
> @@ -35,6 +35,7 @@ if "%1" == "help" (
>         echo.  linkcheck  to check all external links for integrity
>         echo.  doctest    to run all doctests embedded in the documentation if enabled
>         echo.  spelling   to check for typos in documentation
> +       echo.  black      to apply the black formatting to code blocks in documentation 
>         goto end
>  )
> 
> @@ -196,4 +197,11 @@ spelling/output.txt.
>         goto end
>  )
> 
> +if "%1" == "black" (
> +       mkdir %BUILDDIR%
> +       for /f "usebackq tokens=*" %%i in (`dir *.txt /s /b ^| findstr /v /c:"_build" /c:"_theme"`) do (
> +               blacken-docs --rst-literal-block %%i
> +       )
> +       echo "Code blocks reformatted"
> +)
> +
> ```

Thanks, added! Does it save the return code of `blacken-docs` to the `black/output.txt`?
 
> Running blacken-docs it creates one diff to this PR:

It seems that you don't use the newest `black` :thinking: .

**Comment by smithdc1** [comment]
> Does it save the return code of `blacken-docs` to the `black/output.txt`?

No. This calls blacken-docs for _every_ file. So only saves it for the final file. I'm not sure how to pass in a list of files on windows. 

> It seems that you don't use the newest `black` :thinking: .

Ahh OK. Quite possibly.

**Comment by felixxm** [comment]
> > Does it save the return code of `blacken-docs` to the `black/output.txt`?
> 
> No. This calls blacken-docs for _every_ file. So only saves it for the final file. I'm not sure how to pass in a list of files on windows.

I think it's fine, we don't use it in CI so we don't need to capture the return code.

**Comment by carltongibson** [comment]
Just a quick confirm that the `make.bat` looks good.

**Comment by pauloxnet** [comment]
I reviewed the 'makefile' commit and LGTM

File: docs/internals/contributing/writing-documentation.txt [hunk 2]
```diff
@@ -258,6 +261,12 @@ documentation:
   also need to define a reference to the documentation for that environment
   variable using :rst:dir:`.. envvar:: <envvar>`.
 
+.. versionchanged:: 4.2
+
+    All code blocks in Django's docs was reformatted with `blacken-docs`_.
```
Comment (carltongibson) [inline docs/internals/contributing/writing-documentation.txt:266]:
```
    All Python code blocks in the Django docs were reformatted with
    `blacken-docs`_.
```

Could go for `s/docs/documentation/` but I think the friendly voice is fine.

**Review by carltongibson — APPROVED** [review]
OK, I think it looks good. (No doubt they'll be tweaks...) 

Thanks @felixxm and @smithdc1! 🎁

File: docs/internals/contributing/writing-documentation.txt (hunk 2 above)
Comment (felixxm) [reply docs/internals/contributing/writing-documentation.txt]: Updated :+1:

**Comment by felixxm** [comment]
@carltongibson @pauloxnet Thanks :+1:

**Comment by felixxm** [comment]
I found two small issues when reviewing changes made by `blacken-docs` and fixed them in the 1st commit.

**Pull request merged**